server/src/main/java/org/elasticsearch/action/search/SearchRequest.java

```diff
@@ -157,6 +159,16 @@ static SearchRequest subSearchRequest(SearchRequest originalSearchRequest, Strin
         return new SearchRequest(originalSearchRequest, indices, clusterAlias, absoluteStartMillis, finalReduce);
     }
 
```

Comment:
You could reuse the `subSearchRequest` above and just add a nullable `minCompatibleShardNode` argument ? That makes sense to me because we need to set `ccsMinimizeRoundtrips` to false there too and these sub search requests are meant for internal use (using the transport client only).